what4/src/What4/Expr/Builder.hs

```diff
@@ -2860,9 +2816,7 @@ sbMakeExpr sym a = do
     modifyIORef' (sbNonLinearOps sym) (+1)
   case appType a of
     -- Check if abstract interpretation concludes this is a constant.
-    BaseBoolRepr | Just b <- v -> return $ backendPred sym b
```

Comment:
On second thought, I think we'd better put this line back in... even though the boolean operations don't really compute much, this value can still matter when inserting/projecting from structures and arrays.